### DISCUSSION TITLE
Feature Request : -- verysilent Parameter

### DISCUSSION THREAD

**Discussion by cgerke** [microsoft/winget-cli#3027] [Ideas]
This parameter would incorporate all three --silent --accept-source-agreements --accept-package-agreements

This would cut 55 chars of a script line resulting in more readable powershell code when scripting winget.

Alternatively --non-interactive could be used to avoid confusion with inno installers.

Thoughts?

**Comment by denelon** [comment]
We need the explicit arguments to clearly ensure users understand what they are agreeing to. Another generic "--accept-all" type of argument would potentially hide things from unwary users using copy-paste.

**Reply by Masamune3210** [reply]
I would argue that if the user is intentionally copying and pasting commands blindly into a terminal that they are either aware of what they are agreeing to or don't care enough to read it in the first place

**Reply by cgerke** [reply]
> We need the explicit arguments to clearly ensure users understand what they are agreeing to. Another generic "--accept-all" type of argument would potentially hide things from unwary users using copy-paste.

I see your point, my proposal was aimed at system admins managing deployments in the enterprise.

Winget now underpins Intune, replacing "store for business" but also will allow us to create our own "auto patch" system. IMHO winget is moreso an enterprise tool than it is an end user tool.

Everyday users are still more likely to be using the store front end.

**Reply by denelon** [reply]
Intune has the advantage of the IT administrator having the authority to agree on behalf of their users. In cases like that, users wouldn't necessarily see anything. The package just gets installed or updated and it's magic for the end user.

The "in-process" COM API is what Intune integrated with so it's a different flavor of integration than PowerShell or the CLI.

**Reply by cgerke** [reply]
Noted.

It's just half baked at the moment within Intune so I never use the "new store" app style.

99% of all my apps need overrides so everything is scripted via powershell as a win32app.

Perhaps if Intune options get expanded more it becomes a non-issue.

**Reply by cgerke** [reply]
BTW I'd be more than happy to live demo a enterprise implementation some time to show how the tool is being used. It might lead to some interesting conversations or you may be able to tell me I'm not using it as it was intended 😂

**Reply by cgerke** [reply]
> Intune has the advantage of the IT administrator having the authority to agree on behalf of their users. In cases like that, users wouldn't necessarily see anything. The package just gets installed or updated and it's magic for the end user.
> 
> The "in-process" COM API is what Intune integrated with so it's a different flavor of integration than PowerShell or the CLI.

Also just on this. I can't see winget being used more by every users than by systems administrators managing the enterprise or power users like DEVS that are probably going to script their winget installs with json etc and be very explicit about what they are doing. Everyday users will stick with the store. So realistically we should be considering winget an enteprise tool.

IMHO
